nilearn/_utils/cache_mixin.py

```diff
@@ -108,15 +102,17 @@ def _safe_cache(memory, func, **kwargs):
         with open(version_file, 'r') as _version_file:
             versions = json.load(_version_file)
 
-    modules = (nibabel, )
     # Keep only the major + minor version numbers
-    my_versions = dict((m.__name__, m.__version__) for m in modules)
+    my_versions = {nibabel.__name__: nibabel.__version__}
     commons = set(versions.keys()).intersection(set(my_versions.keys()))
-    collisions = [
-        m for m in commons if not _compare_version(
-            versions[m], '==', my_versions[m]
-        )
-    ]
+    try:
+        collisions = [
+            m for m in commons if not _compare_version(
+                versions[m], '==', my_versions[m]
+            )
+        ]
+    except TypeError:
+        collisions = ['']
```

Comment:
wouldn't "collisions = ['nibabel']" be a bit clearer?

Reply:
but still it is a bit weird to have this set logic with intersections and iterating over sequences if this is just for nibabel we could just have something like `if "nibabel" in versions and versions_differ(versions["nibabel"], nibabel.__version__): flush_cache()`.

also note that if we use nilearn once with CHECK_CACHE_VERSION cleared, we can end up with a cache directory containing a json listing a nibabel version which actually isn't the one used to create the data in the cache (this function does not flush the data but overwrites the versions JSON). and then future checks will be based on false information, or information that is true for part of the results in the cache and false for some other results

Reply:
The best way to simplify it was to create another function to handle the exception because it made it hard to avoid repeating code.

And I added a return statement to avoid the behavior you describe.